### PR TITLE
100998 - check if user exists in local repository before checking cache

### PR DESCRIPTION
**Pull request by janid1967**
... which get user from ProCoSys

[AB#100998](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/100998)

### CODE REVIEW

**Comment by jsoi-eq** [comment]
Linked to wrong AzureDev Ops card? 100998 vs 100989?

**Comment by janid1967** [comment]
> 

You're right. Fixed

File: src/tests/Equinor.ProCoSys.Auth.Tests/Authorization/ClaimsTransformationTests.cs [hunk 1]
```diff
@@ -168,6 +173,7 @@ public async Task TransformAsync_ShouldNotAddAnyClaims_ForPrincipalWithoutOid()
         [TestMethod]
         public async Task TransformAsync_ShouldNotAddAnyClaims_WhenPersonNotFoundInProCoSys()
         {
+            _localPersonRepositoryMock.Setup(p => p.ExistsAsync(Oid)).ReturnsAsync(false);
```
Comment (jsoi-eq) [inline src/tests/Equinor.ProCoSys.Auth.Tests/Authorization/ClaimsTransformationTests.cs:176]: Should we add unit tests for all permutations for GetRestrictionRoleClaimValue, so in addition to true/true and false/false tests which already exists, should we add tests for true/false and false/true?

Comment (janid1967) [reply src/tests/Equinor.ProCoSys.Auth.Tests/Authorization/ClaimsTransformationTests.cs:176]: I fix

**Pull request merged**